src/atomicassets.cpp

```diff
@@ -503,6 +503,43 @@ ACTION atomicassets::locktemplate(
 }
 
 
+/**
+* Reduces the max supply of the template to the new max supply
+* This means that afterwards, NFTs can only be minted up to the new max supply
+* @required_auth authorized_editor, who is within the authorized_accounts list of the collection
+**/
+
+ACTION atomicassets::redmaxtemplt(
+    name authorized_editor,
+    name collection_name,
+    int32_t template_id,
+    uint32_t new_max_supply
+) {
+    require_auth(authorized_editor);
+
+    auto collection_itr = collections.require_find(collection_name.value,
+        "No collection with this name exists");
+
+    check_has_collection_auth(
+        authorized_editor,
+        collection_name,
+        "The editor is not authorized within the collection"
+    );
+
+    templates_t collection_templates = get_templates(collection_name);
+    auto template_itr = collection_templates.require_find(template_id,
+        "No template with the specified id exists for the specified collection");
+
+    check(new_max_supply > 0, "Can't set the max supply of a template to 0");
+
+    check(template_itr->max_supply > new_max_supply,"Can't increase the max supply of a template");
```

Comment:
This doesn't cover the case where the current max supply is 0 (meaning unlimited).
Also I would suggest a slight rephrasing of the error message, since it's not only that you can't increase it, you also can't keep it the same. I suggest this:

`check(template_itr->max_supply == 0 || template_itr->max_supply > new_max_supply, "The new max supply must be lower than the existing max supply");`

Reply:
Current max supply > new max supply > 0,

so it prohibits the 0 case, although the error could be made more verbose.

I do see your point about enabling defined locking for currently infinite max supply templates, will be addressed shortly

Reply:
Yeah I see that the 0 case is prohibited, my point is that I don't think it should be. setting a max supply for a previously unlimited template seems reasonable.